### PR TITLE
Aldldl playerdata

### PR DESCRIPTION
**Pull request by aldldl**
Redid playerdata after a rename and change of AKA's and permissions. Please check this new PR as it should be much better. Still may need some help on the cache though.

### CODE REVIEW

File: infoBot.js [hunk 1]
```diff
@@ -130,9 +130,25 @@ client.on("message", async msg => {
...
         if (!allowed) {
-            console.log(`Unauthorized `);
-            msg.channel.send(`You do not have ${role.name} permission.`);
-            return;
+            //if(command.name == `playerdata` && args[0] == msg.member.displayName || command.name == `playerdata2` && args[0] == msg.member.displayName || command.name == `playerdata2` && !args[0])
+            //            {
+            //                console.log(`Self look up by ${msg.member.displayName}`);
+            //            } // exception for self lookup
+            //        else    
+            //            {
```
Comment (aldldl) [inline infoBot.js]:
```suggestion
```

File: infoBot.js [hunk 2]
```diff
@@ -130,9 +130,25 @@ client.on("message", async msg => {
...
+            //                {
+            //                    msg.channel.send(`${roleError1} Nor do you seem to be looking up your own data.`);
+            //                }
+            //                else
+            //                {
+                                msg.channel.send(`You do not have ${role.name} permission.`);
+            //                }
+                            return;
+            //            }
+
```
Comment (aldldl) [inline infoBot.js]:
```suggestion
```

File: infoBot.js [hunk 3]
```diff
@@ -130,9 +130,25 @@ client.on("message", async msg => {
         let role = await msg.guild.roles.fetch(req_role)
         let allowed = msg.member.roles.highest.comparePositionTo(role) >= 0;
         if (!allowed) {
```
Comment (aldldl) [inline infoBot.js:132]:
```suggestion
        if (!allowed) {
        console.log(`Unauthorized `);
            msg.channel.send(`You do not have ${role.name} permission.`);
            return;
            }
```

File: commands/discordpin.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const { Role } = require("discord.js");
```
Comment (tovernaar123) [inline commands/discordpin.js:1]: ? why are you requiring role from discord.js?

File: commands/discordpin.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+const Discord = require('discord.js');
```
Comment (tovernaar123) [inline commands/discordpin.js:2]: also see no reason to need a ref to discord

File: commands/discordpinun.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+const { Role } = require("discord.js");
```
Comment (tovernaar123) [inline commands/discordpinun.js:1]: same no roles object in the discord module

File: commands/discordpinun.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+const Discord = require('discord.js');
```
Comment (tovernaar123) [inline commands/discordpinun.js:2]: same you are not using this.

File: commands/playerdata.js [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+            //if (1 == 1) 
+            //    { // start PHIDAS format feature
+
+                    
+                    function ts(x) { //thousands separator
+                        if (x === undefined) {
+                            return 0;
+                        } else {
+                            try {
+                                return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
```
Comment (tovernaar123) [inline commands/playerdata.js:17]: the try catch here shood really not be needed

File: commands/playerdata.js [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+                            'Deaths':'Deaths',
+                            'DamageDealt':'Damage Dealt',
+                            'DistanceTravelled':'Distance Travelled',
+                            'CapsulesUsed':'Capsules Used',
+                            'EntityRepaired':'Machines Repaired',
+                            'DeconstructionPlannerUsed':'Decon Planner Used',
+                            'MapTagsMade':'Map Tags Created'
+                        };
+                        
+                        let msg_4 = [];
```
Comment (tovernaar123) [inline commands/playerdata.js:53]: please name you variables as to what they are msg_4 is not  very descriptive

File: commands/playerdata.js [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+                    
+                        for (let i = 0; i < layout.length; i++) {
+                            let msg_3 = [];
+                            
+                            if (layout[i] == 'Playtime' || layout[i] == 'AfkTime') {
+                                try {
+                                    var t1 = msg_2[layout[i]];
```
Comment (tovernaar123) [inline commands/playerdata.js:60]: Why a try catch?

Comment (tovernaar123) [reply commands/playerdata.js:60]: also what is t1 change the name of this variable

File: commands/playerdata.js [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+                            'CapsulesUsed':'Capsules Used',
+                            'EntityRepaired':'Machines Repaired',
+                            'DeconstructionPlannerUsed':'Decon Planner Used',
+                            'MapTagsMade':'Map Tags Created'
+                        };
+                        
+                        let msg_4 = [];
+                    
+                        for (let i = 0; i < layout.length; i++) {
+                            let msg_3 = [];
```
Comment (tovernaar123) [inline commands/playerdata.js:56]: what is msg_3 please change the name of this variable to smt more understandable.

File: commands/playerdata.js [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+                            
+                            if (layout[i] == 'Playtime' || layout[i] == 'AfkTime') {
+                                try {
+                                    var t1 = msg_2[layout[i]];
+                                } catch (e) {
+                                    var t1 = 0;
+                                }
+                                
+                                if (isNaN(t1)) {
```
Comment (tovernaar123) [inline commands/playerdata.js:65]: why would this be nan?

File: commands/playerdata.js [hunk 13]
```diff
@@ -0,0 +1,227 @@
...
+                                    msg_3.push(layout_dict[layout[i]]);
+                                    msg_3.push(h1 + ' h ' + m1 + ' m');
+                                } else {
+                                    msg_3.push(layout_dict[layout[i]]);
+                                    msg_3.push(m1 + ' m ');
+                                }
+                    
+                            } else {
+                                try {
+                                    msg_3.push(layout_dict[layout[i]]);
```
Comment (tovernaar123) [inline commands/playerdata.js:82]: again no reason for a try catch, just test if layout[i] exsits and then if layout_dict[layout[i]].

File: commands/playerdata.js [hunk 14]
```diff
@@ -0,0 +1,227 @@
...
+                                }
+                            }
+                    
+                            msg_4.push([msg_3[0], msg_3[1]]); // push title and value in array
+                        }
+                        
+                        return msg_4;
+                    }
+                    //Find "items" from datastore 
+                    const fs  = require('fs');
```
Comment (tovernaar123) [inline commands/playerdata.js:96]: put all require on the top of the file

File: commands/playerdata.js [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+                    let checkdata = mydata["PlayerData"][key1];
+                    let finaldata;
+                    //Checks to see if any data was retured at all, if the name is not in the database, or the database is not accessable than it will return an error and stop running the command
+                    if(!checkdata)
+                        {
+                            msg.channel.send('Name error: Name not found in datastore. Check the name or try again later.');console.log(`Name Not Found`); return;
+                        }
+                    //if it didnt stop based on the name not returining it will then filter out only the Statistics (removing prefrences like alt mode, join msg etc)
+                    finaldata = mydata["PlayerData"][key1]["Statistics"];
+                    // set item to equal finaldata - to make Phidas formatting as plug and play as possible - not really needed otherwise
```
Comment (tovernaar123) [inline commands/playerdata.js:113]: could also just be checkdata.Statistics.

File: commands/playerdata.js [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+                    item = finaldata;
```
Comment (tovernaar123) [inline commands/playerdata.js:114]: why are you renaming variables?

File: commands/playerdata.js [hunk 17]
```diff
@@ -0,0 +1,227 @@
...
+
+                    let channel = msg.channel;
```
Comment (tovernaar123) [inline commands/playerdata.js:116]: again what is the point of giving this a different name

File: commands/playerdata.js [hunk 18]
```diff
@@ -0,0 +1,227 @@
...
+                    let result = profile(item);
+                    let lookup = name;
```
Comment (tovernaar123) [inline commands/playerdata.js:118]: same why does this need onther name

File: commands/playerdata.js [hunk 19]
```diff
@@ -0,0 +1,227 @@
...
+                    //log and channel (nonembed) messages sent
+                    console.log(`Player Data Requested by ${msg.member.displayName}\nUsername: ${lookup} `);
+                    channel.send(`\`\`\`txt\nPlayer Data Requested by ${msg.member.displayName}\nUsername: ${lookup}\n\`\`\``);
+                    //set up embeds
+                    const Embed = new Discord.MessageEmbed();
+                    Embed.setColor("0x00ff00");
+
+                    for (let i = 0; i < result.length/2; i+=2) {
+                        try {
+                            Embed.addField(result[i][0], result[i][1], true);
```
Comment (tovernaar123) [inline commands/playerdata.js:129]: please dont use try catch like this jus test if the array’s have the elments you want to use if not dont acces them

File: commands/playerdata.js [hunk 20]
```diff
@@ -0,0 +1,227 @@
...
+                        } catch (e) {
+                            Embed.addField(`** **`, `** **`, true);
+                        }
+
+                        try {
```
Comment (tovernaar123) [inline commands/playerdata.js:134]: again no real reason for a try catch.

File: commands/playerdata.js [hunk 21]
```diff
@@ -0,0 +1,227 @@
...
+                            Embed.addField(result[i+1][0], result[i+1][1], true);
+                        } catch (e) {
+                            Embed.addField(`** **`, `** **`, true);
+                        }
+
+                        Embed.addField(`** **`, `** **`, true);
+                    }
+
+                    const Embed2 = new Discord.MessageEmbed();
```
Comment (tovernaar123) [inline commands/playerdata.js:143]: please state what this really is, is this the second embed that will be sent?

File: commands/playerdata.js [hunk 22]
```diff
@@ -0,0 +1,227 @@
...
+                    Embed2.setColor("0x00ff00");
+
+                    for (let i = 0; i < result.length/2; i+=2) {
+                        let j = i + result.length/2;
+                        try {
```
Comment (tovernaar123) [inline commands/playerdata.js:148]: again why a try catch?

File: commands/playerdata.js [hunk 23]
```diff
@@ -0,0 +1,227 @@
...
+                            Embed2.addField(result[j][0], result[j][1], true);
+                        } catch (e) {
+                            Embed2.addField(`** **`, `** **`, true);
+                        }
+
+                        try {
```
Comment (tovernaar123) [inline commands/playerdata.js:154]: stop with all of these ty catch blocks :P .

File: commands/playerdata.js [hunk 24]
```diff
@@ -0,0 +1,227 @@
...
+
+
+module.exports = {
+    name: 'playerdata',
+    aka: ['pd','pdformatted','pdnice','playerdataformatted','userdata'],
+    description: 'Get stats (datastore info) for youself (all users) or any users (Board+)',
+    guildOnly: true,
+    args: false,
+    helpLevel: "all",
+    // required_role: role.board,
```
Comment (tovernaar123) [inline commands/playerdata.js:190]: why is the not used?

File: commands/playerdata.js [hunk 25]
```diff
@@ -0,0 +1,227 @@
...
+    usage: ` <name>`,
+    execute(msg, args, _, internal_error) {
+        
+        async function runCommand()
+
```
Comment (tovernaar123) [inline commands/playerdata.js:195]: what does this function do?

File: commands/playerdatajson.js [hunk 26]
```diff
@@ -0,0 +1,54 @@
...
+        //this command unlike its brother pd2 DOES require a name to be specified. You can still look up yourself but you must add in your username to look up.
+        if (!name) 
+        {
+            msg.channel.send(`Please pick a name first. Just the name - CAPS COUNT (use \`playerdata2\` if you dont want to add in the name`)
+                .catch((err) => { internal_error(err); return });
+            return;
+        }
+        if (name) 
+        {
+            const fs  = require('fs');
```
Comment (tovernaar123) [inline commands/playerdatajson.js:24]: put it on the top.

File: commands/playerdatajson.js [hunk 27]
```diff
@@ -0,0 +1,54 @@
...
+            let rawdata = fs.readFileSync('/home/exp_admin/api_v2/persistent_storage.json');
+            let mydata = JSON.parse(rawdata);
+            key1 = `${name}`; // in dataFile
+            let checkdata = mydata["PlayerData"][key1];
```
Comment (tovernaar123) [inline commands/playerdatajson.js:28]: change key1 to name. and then remove key1 = name

File: rcon_auto_connect.js [hunk 28]
```diff
@@ -15,28 +15,33 @@ exports.rcon_connect = async function(port, i) {
         password: rconpw,
     });
 
+    let Iconnected = false;
```
Comment (tovernaar123) [inline rcon_auto_connect.js:18]: i mean its fine by me but what does the I stand for?

**Review by tovernaar123 — CHANGES_REQUESTED** [review]
Lots of random try catches that make the code not so readable.

**Comment by tovernaar123** [comment]
this shood not be merged to master and has already been put in dev

**Pull request closed without merging**